src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75F;
+    private int threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    private Node<K, V>[] table = new Node[DEFAULT_CAPACITY];
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            size = 0;
+            threshold *= 2;
+            resize();
+        }
+        int keyIndex = hash_code(key) % table.length;
```

Comment:
it's better if you create a separate method for getting indexes, for instance, getIndex(), because your code is duplicated here